### PR TITLE
Make property value accept types other than string

### LINKED ISSUE #67: Make the property value accept types other than string.
## Feature Request

**Is your feature request related to a problem? Please describe.**  

Currently, only the value of string is received as value, so it is cumbersome to use other types of information.

**Describe the solution you'd like**  

It must be implemented to receive primitive types such as boolean and int.

**Describe alternatives you've considered**  

**Additional context**

### PR DESCRIPTION
**Pull request by HyeockJinKim**
# Summary 
Expanded to receive not only string type but also number, boolean,
and date types.

- It closes #67

## How to Review? 

- Please refer to my self-review submitted as the review.

### CODE REVIEW

File: packages/message/src/type/stringToDataType.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        return new Date(JSON.parse(value));
+      }
+      return value;
+    case PropertyDataType.Boolean:
+      if (typeof value === 'string') {
+        return JSON.parse(value);
+      }
+      return value;
+  }
+}
```
Comment (HyeockJinKim) [inline packages/message/src/type/stringToDataType.ts:23]: Each data type converted to string is changed to an appropriate data type.

File: packages/message/src/type/index.ts [hunk 2]
```diff
@@ -1,27 +1,4 @@
...
-
-export type Signature = {
-  authType: AuthType;
-  keyType: KeyType;
-  value: string;
-};
+export * from './stringToDataType';
+export * from './type';
+export * from './typeOfObject';
+export * from './typeOfProperty';
```
Comment (HyeockJinKim) [inline packages/message/src/type/index.ts:4]: Now, index.ts only do export * from `./*`

File: packages/message/src/type/typeOfProperty.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    case 'number':
+      return PropertyDataType.Number;
+    case 'boolean':
+      return PropertyDataType.Boolean;
+    case 'object':
+      return typeOfObject(value);
+    default:
+      throw new Error('Unreachable Code in typeOfProperty.');
+  }
+}
```
Comment (HyeockJinKim) [inline packages/message/src/type/typeOfProperty.ts:17]: Check type and return `PropertyDataType`.

File: packages/message/src/utils/utf8ToBase64.ts [hunk 4]
```diff
@@ -1,3 +1,11 @@
-export function utf8ToBase64(str: string): string {
-  return Buffer.from(encodeURIComponent(str), 'binary').toString('base64');
+import { PropertyData } from '@0auth/message';
+
+export function utf8ToBase64(value: PropertyData): string {
+  if (typeof value === 'string') {
+    return Buffer.from(encodeURIComponent(value), 'binary').toString('base64');
+  }
+  return Buffer.from(
+    encodeURIComponent(JSON.stringify(value)),
```
Comment (HyeockJinKim) [inline packages/message/src/utils/utf8ToBase64.ts:8]: If type of value is not string, do `JSON.stringify` and get string value .

File: packages/message/test/message.spec.ts [hunk 5]
```diff
@@ -49,18 +55,101 @@ describe('test message utils', () => {
...
+    type: PropertyType.Raw,
+    value: true,
+  };
+
+  it('test stringToDataType of string type', () => {
+    const stringValue = stringProperty.value;
+    expect(stringToDataType(stringValue, PropertyDataType.String)).to.be.equal(
+      stringValue,
+    );
+  });
```
Comment (HyeockJinKim) [inline packages/message/test/message.spec.ts:108]: test for `stringToDataType`.

File: packages/server/src/issuer/authProperty.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 import { Predicate, RegisterInfo, SecretKey } from '../type';
-import { authByAuthType } from './authByAuthType';
+import { authByAuthType } from '@0auth/server';
 
 export function authProperty(properties: Property[]): RegisterInfo {
   let isPassed = true;
   const propertiesObj = propertyObject(properties);
   return {
-    validate(key: string, func: Predicate<string>): RegisterInfo {
+    validate(key: string, func: Predicate<unknown>): RegisterInfo {
```
Comment (HyeockJinKim) [inline packages/server/src/issuer/authProperty.ts:10]: Now, the validate function receives only functions or lambdas with no parameter type specified.

**Review by HyeockJinKim — COMMENTED** [review]
self review

**Review by anthonyminyungi — APPROVED** [review]
Looks good. I'll go merge this.

**Pull request merged**